src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs

```diff
@@ -903,11 +905,14 @@ private ISet<string> CalculateFilesToExcludeInPack(IPackTaskRequest<IMSBuildItem
 
         private static void InitializePackageDependencies(
             LockFile assetsFile,
+            PackageBuilder packageBuilder,
             Dictionary<NuGetFramework, HashSet<LibraryDependency>> dependenciesByFramework,
             ISet<NuGetFramework> frameworkWithSuppressedDependencies)
         {
+            StringBuilder nowarnProperties = new();
```

Comment:
Rather than make these into a complex string and parse later, can they just be kept in an object and used later?  hopefully that's not a dumb question, I'm not very familiar with where they go between here and there.

Reply:
I added new property to packagebuilder instead of using string dictionary, please check now.